Scripts/DownloadInitialResources/DownloadInitialResourcesScript.swift

```diff
@@ -0,0 +1,405 @@
+//
+//  DownloadInitialResourcesScript.swift
+//  godtools
+//
+//  Created by Levi Eggert on 8/18/22.
+//  Copyright © 2022 Cru. All rights reserved.
+//
+
+import Foundation
+import Combine
+
+@main
+enum DownloadInitialResourcesScript {
+             
+    private static var runScriptCancellable: AnyCancellable?
+    
+    static func main() {
+                
+        let apiBaseUrl: String = "https://mobile-content-api.cru.org"
+        let session: URLSession = getIgnoreCacheSession()
+        
+        let semaphore = DispatchSemaphore( value: 0)
+        
+        DownloadInitialResourcesScript.runScriptCancellable = createTemporaryAssetsDirectory()
+            .flatMap({ result -> AnyPublisher<(resourcesData: Data, languagesData: Data), Error> in
+                
+                return getResourcesPlusLatestTranslationsAndAttachmentsPlusLanguagesData(
+                    session: session,
+                    apiBaseUrl: apiBaseUrl
+                )
+            })
+            .flatMap({ result -> AnyPublisher<(resourcesData: Data, languagesData: Data), Error> in
+                
+                return saveResourcesAndLanguagesDataToBundleAsJsonFile(
+                    resourcesData: result.resourcesData,
+                    languagesData: result.languagesData
+                )
+            })
+            .flatMap({ result -> AnyPublisher<ResourcesPlusLatestTranslationsAndAttachmentsModel, Error> in
+                
+                return parseResourcePlusLatestTranslationsAndAttachmentsModel(resourcesData: result.resourcesData)
+            })
+            .flatMap({ result -> AnyPublisher<[Bool], Error> in
+                
+                return saveResourceAttachmentsToBundle(resourcesPlusLatestTranslationsAndAttachments: result)
+            })
+            .flatMap({ result -> AnyPublisher<String, Error> in
+                                
+                return deleteTemporaryAssetsDirectory()
+            })
+            .sink(receiveCompletion: { result in
+                semaphore.signal()
+            }, receiveValue: { result in
+                
+            })
+        
+        semaphore.wait()
```

Comment:
I don't think I've ever seen a semaphore actually in use before.  Does this just ensure that nothing else will get executed until this script finishes?

Reply:
I'm going to see if I can write some unit tests to ensure those assets are persisted to the app bundle too.

Reply:
Hey @rachaelblue for testing this you will want to make sure you can run that script and then run the app without a network connection to see if the resources, languages, and attachments persist.

What you can do is:
1. Ensure Build Phases > Download Initial Resources > For install builds only is unchecked.  Run the a new build with a network connection so the Download Initial Resources script runs and attaches to the device then quit the app.
2. Disable network connection and run the app and see if the lessons, all tools, favorited tools are populated with data and attachment banners are visible on those tool cards. (I need to fix something here because I noticed the lessons and all tools lists aren't loading.  Most likely due to an observable not firing off in InitialDataDownloader when the network request fails due to no network connection.)

Reply:
If you are looking for another example, I use a semaphore to provide external synchronization for Snowplow events on Android.

relevant code: https://github.com/CruGlobal/android-gto-support/blob/master/gto-support-snowplow/src/main/kotlin/org/ccci/gto/android/common/snowplow/events/EventSynchronizer.kt#L11

Reply:
Hey @rachaelblue just pushed 1 commit which addresses the issue of those observables not firing off from InitialDataDownloader in the event the network request fails.   You should be able to test now.